### ISSUE TITLE
"No such file or directory" when rolling log file on xctest

### ISSUE THREAD

**Issue by Diarrhio** [CocoaLumberjack/CocoaLumberjack#1203]
### New Issue Checklist

* [X] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [X] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [X] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 14.3
 CocoaLumberjack Version | 3.7
 Integration Method      | cocoapods
 Xcode Version           | Xcode 12.3
 Repro rate              | all the time (100%)
 Demo project link       | 

### Issue Description and Steps

I have Cocoapods embedded in a framework which is being unit tested in a standard XCTest (non-UI). Our framework has a FileLogger attached like so:

```
    let fileLogger = DDFileLogger()
    fileLogger.rollingFrequency = 60 * 60 * 24 // 24 hours
    fileLogger.logFileManager.maximumNumberOfLogFiles = 7
    fileLogger.doNotReuseLogFiles = true
    DDLog.add(fileLogger)
```

Our framework is set to verbose logging and outputs a lot of logs as our unit tests are running. When CocoaLumberjack goes to roll a log file, I see the following output in the console:

```
DDLogFileInfo: setxattr(lumberjack.log.archived, /Users/dario/Library/Developer/CoreSimulator/Devices/4D44267D-7B5F-483E-A8FE-597D883B6D57/data/Library/Caches/Logs/com.apple.dt.xctest.tool 2021-01-27--12-45-44-052.log): error = No such file or directory
```

The call stack is:
```
#0	0x000000010d62cfa1 in -[DDLogFileInfo addExtendedAttributeWithName:] at /Users/dario/Dev/Builds/MyApp/Pods/CocoaLumberjack/Sources/CocoaLumberjack/DDFileLogger.m:1675
#1	0x000000010d62c297 in -[DDLogFileInfo setIsArchived:] at /Users/dario/Dev/Builds/MyApp/Pods/CocoaLumberjack/Sources/CocoaLumberjack/DDFileLogger.m:1481
#2	0x000000010d627819 in -[DDFileLogger lt_rollLogFileNow] at /Users/dario/Dev/Builds/MyApp/Pods/CocoaLumberjack/Sources/CocoaLumberjack/DDFileLogger.m:893
#3	0x000000010d629a34 in __58-[DDFileLogger lt_monitorCurrentLogFileForExternalChanges]_block_invoke at /Users/dario/Dev/Builds/MyApp/Pods/CocoaLumberjack/Sources/CocoaLumberjack/DDFileLogger.m:1125
```

The file it is trying to set the attribute for does not in fact exist at that location on disk.

In `lt_rollLogFileNow`, I see that `_currentLogFileInfo` is mostly nil except for the `filePath` property. Not sure if that is expected at that point. However, I only see `addExtendedAttributeWithName` referencing `filePath` and no other properties, so it's probably ok. Seems strange that all the properties on it would be nil though

**Comment by ffried** [comment]
`lt_monitorCurrentLogFileForExternalChanges` hints at an external modification of the log file. In this case likely an external delete. 
This can have multiple reasons in the context of unit tests. E.g. when running tests in parallel, it might be that the simulator cleans up.
Or - depending on your unit tests - there are tests that deletes the log files (e.g. in setUp or tearDown)?

Either way, it's nothing serious. We can't really set an extended attribute on an already deleted file. 🙂 
Also, only `filePath` being non-nil in this case makes sense, given that all other properties of `DDLogFileInfo` are being lazily from the file. So if no one accessed those properties, the backing vars will be nil. Also, since the file does not exist, even if they are accessed, they can't be read and thus are nil.

We should maybe check for the error and log with a different level if the file simply does not exist anymore.

**Comment by Diarrhio** [comment]
Thanks for the quick reply. We don't have parallel test execution happening on our tests, so I think that rules that out. Our setUp/tearDown implementations only delete the top level object declared in our framework, the one that adds the FileLogger. It doesn't do anything with `DDLog` other than adding the file logger early on in its constructor. In its `deinit`, method it simply does a DDLogDebug to log that its been destroyed. There are no calls to `DDLog.removeLogger` anywhere in our codebase as it's just something we never really want to do.

Anyways, on our side it's harmless (though I haven't carefully scoured production app logs to see if we're seeing the error message there. It does tend to pollute our XCTest console logs though, which is a bit annoying, but certainly no deal breaker.

**Comment by ffried** [comment]
I've opened a small PR that should fix this.

**Comment by Diarrhio** [comment]
Many thanks :)
